### ISSUE TITLE
Double click on citation should open the LaTeX editor at the respective place

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#11996]
We have the push-to-application functionality. We also have the ["LaTeX citations" functionality](https://docs.jabref.org/advanced/entryeditor/latex-citations). It would be nice if I as user could navigate to the listed TeX file in the LaTeX editor by a double click.

---

The resulting code will add a method `jumpToLine(Path fileName, int line, int column)` to `org.jabref.gui.push.PushToApplication`.

Add `default` empty implementation, to avoid compile errors at the beginning.

Hints

2. Leave out configuration of each tool; reuse org.jabref.gui.push.PushToApplicationPreferences and hard-code the parameters to jump to a line
3. Implement for `TeXstudio`. `texstudio --line "$lineno" "$file"`
4. The `commandPath` contains the path tot he executable.
5. Optionally: implement for more applications (e.g., `emacsclient +4:3 FILE`)
6. Optionally: Add VSCode as new push target

Note: The push to applications functionality is described at https://docs.jabref.org/cite/pushtoapplications.

---

Example:

1. Open `src\test\resources\pdfs\IEEE\ieee-paper.tex`
2. Double click on "Kopp et al."
3. Select tab "LaTeX ciations" in entry editor
4. Double click on the first entry in the table:\
    ![Image](https://github.com/user-attachments/assets/8dd455c1-7e56-4d19-93de-d7762fe550b7)
5. See that nothing happens

Instead of 5, TeXstudio should be opened with `ieee.tex` opened and jumping to line 26.

**Comment by u7465990** [comment]
I’m currently working on a university assignment focused on contributing to open source projects. Could I take this issue? I would be happy to work on it and contribute to the project.

Thank you !

**Comment by koppor** [comment]
> Could I know what function is called after clicking the push to external file?

What do you mean? The "push to applications" functionality? It is not directly related to this wish - you can only learn how things are implemented for a different use case.

You did not provide any context or test application - thus it makes difficult to point you to concrete classes.

Start reading code at `org.jabref.gui.push.AbstractPushToApplication#pushEntries(org.jabref.model.database.BibDatabaseContext, java.util.List<org.jabref.model.entry.BibEntry>, java.lang.String, java.lang.ProcessBuilder)`. You can use Ctrl+click on a method to see where it is defined or where it is called.

> This entry is called LatexCitationsTab. This tab is a java class that us javafx to set the preferences. But I don't what happened after the user double click it.

Curerntly nothing. It is your task to add this functionality. See step 5 in the issue description (the part at "Example:")

> How could I found the tex file path in this latex citations tab ?

`org.jabref.model.texparser.Citation` is the object holding the `Path`. It is used at `org.jabref.gui.texparser.CitationsDisplay`. `CitationsDisplay` displays the citation.

> I think this issue is about open the external tex editor. 

Not quite. It is to instruct the opened editor to JUMP to a lline.

It seems you never used LaTeX and BibTeX.

Here two slides, maybe, they help to understand the goal:

![Image](https://github.com/user-attachments/assets/1f603711-cb59-429e-b361-8251a46f5495)

![Image](https://github.com/user-attachments/assets/fd6fb5b7-1f52-491b-8f6d-e978b6082989)

Imagine now that someone writes a book and has several .tex files. That person uses JabRef for editing the .bib file. Now, they want to have the text edtor jump to the .tex file where the current citation is used.

> The commandPath is setted in the Preference. 

Retrieve it similarly to `preferences.getPushToApplicationPreferences()`.

> The path and the line number is setted in the latex citations.

It is available in `org.jabref.model.texparser.Citation`.

**Comment by u7465990** [comment]
Sorry. I didn't provided the example. I read the pushEntry function and try to understand what happened. 

I did some test cases. I open the 'ieee-paper.bib' in 'src\test\resources\pdfs\IEEE\'. I select the first article and press the push the button 'Push entry to external application'. Then the 'pushEntries' is called and then the command 'texstudio --insert-cite \cite{Alver2007a}' is runned by terminal. 

And in this issue, the pushEntries will run 'texstudio --line "$lineno" "$file"' after double clicking the the first entry in the table. But currently after clicking the entry, the pushEntry is not runned. How could I improved it?

**Comment by koppor** [comment]
> And in this issue, the pushEntries will run 'texstudio --line "$lineno" "$file"' after double clicking the the first entry in the table. But currently after clicking the entry, the pushEntry is not runned. How could I improved it?

It is exactly the task of this issue to implement running `texstudio --line "$lineno" "$file"` on double click of a citation.

**Comment by u7465990** [comment]
How to catch the doulble click? Should I use the javaFx event listener in the LatexCitationTabs?

**Comment by koppor** [comment]
> How to catch the doulble click? Should I use the javaFx event listener in the LatexCitationTabs?

I do not know. I would need to try out, too.

Searching for "double click" reveals

```java
    private void handleItemMouseClick(LinkedFileViewModel linkedFile, MouseEvent event) {
        if (event.getButton() == MouseButton.PRIMARY && (event.getClickCount() == 2)) {
            // Double click -> open
            linkedFile.open();
        }
    }
```

You need to handle it in `CitationsDisplay`. Maybe StackOverflow can help you https://stackoverflow.com/q/9722418/873282 ?

**Comment by u7465990** [comment]
I found the problem. The 'LatexCitationTab' use 'GidPane' to divided schema. Those citations are actually one list in a 'VBox' of 'GridPane'. I found that list works as a 'ReadOnlyListWrapper' but it should works like the table of entries for articles, do nothing while click once and do some actions while double click. But I didn't found the class of 'DoubleClickOnlyListWrapper'. How should I fix it?
